bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -306,9 +306,9 @@ void PreProcessor::onMessage<PreProcessRequestMsg>(PreProcessRequestMsg *msg) {
       return;
     }
   }
-  registerRequest(ClientPreProcessReqMsgUniquePtr(), preProcessReqMsg);
-  // Pre-process the request, calculate a hash of the result and send a reply back
-  launchAsyncReqPreProcessingJob(preProcessReqMsg, false, false);
+  if (registerRequest(ClientPreProcessReqMsgUniquePtr(), preProcessReqMsg))
+    // Pre-process the request, calculate a hash of the result and send a reply back
+    launchAsyncReqPreProcessingJob(preProcessReqMsg, false, false);
```

Comment:
minor - I think it's always safer to have separate {...} blocks for if-then-else statements